### PR TITLE
add amsmath extension

### PR DESCRIPTION
**Pull request by chrisjsewell**
Adds extension to specifically capture top-level [amsmath](https://ctan.org/pkg/amsmath?lang=en) latex environments

cc @choldgraf @phaustin for comment

### CODE REVIEW

**Comment by phaustin** [comment]
@chrisjsewell do I need to do anything special to activate the extension from myst_nb?

**Comment by chrisjsewell** [comment]
> do I need to do anything special to activate the extension from myst_nb?

Few more steps until you can use it in myst-nb I'm afraid: https://github.com/executablebooks/MyST-NB/issues/126#issuecomment-622935821

**Comment by phaustin** [comment]
this set  looks great -- is there a reason that the vertical, bracketed and parenthetical matrix environments (bmatrix, pmatrix, vmatrix) are missing?  Those would be at the top of the next tier of my own list of most commonly used mathjax environments from  the bottom section   of https://docs.mathjax.org/en/v2.7-latest/tex.html

File: markdown_it/extensions/amsmath/__init__.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+# other "non-top-level" environments:
+
+# 3.4 the split environment is for single equations that are too long to fit on one line
+# and hence must be split into multiple lines,
+# it is intended for use only inside some other displayed equation structure,
+# usually an equation, align, or gather environment
+
+# 3.7 variants gathered, aligned,and alignedat are provided
+# whose total width is the actual width of the contents;
+# thus they can be used as a component in a containing expression
```
Comment (chrisjsewell) [inline markdown_it/extensions/amsmath/__init__.py:33]: In response to your question: http://anorien.csc.warwick.ac.uk/mirrors/CTAN/macros/latex/required/amsmath/amsldoc.pdf states that these should not be top-level environments. Have you found a use case where they are?

**Comment by chrisjsewell** [comment]
> bmatrix, pmatrix, vmatrix

Ah yes, I missed these in section 4.1 of http://anorien.csc.warwick.ac.uk/mirrors/CTAN/macros/latex/required/amsmath/amsldoc.pdf. That's why I asked for your comment 😄

**Comment by phaustin** [comment]
just to flag the comment:

```
# eqnarray is another math environment, but is not part of amsmath,
# and it is better to use align or equation+split instead
```
Note that eqnarray is the default way to write systems of equations in latex.  It's definitely not recommended -- i.e.  http://tug.org/TUGboat/tb33-1/tb103madsen.pdf, but unfortunately it's going to be present in a lot of jupyter notebooks, and in all standard latex manuals.  AMS evidently decided that simply deprecating it in mathjax wasn't worth the struggle -- I think forcing users to take their medicine isn't a bad thing, but I'd anticipate a lot of bug reports from new users, which will need to be redirected to a standard doc snippet.

**Comment by phaustin** [comment]
One other thing, I see flalign on your list, but it doesn't look like it's an environment either mathjax or katex support.  That would be something that would work for the latex builder, but revert to regular align for html?

**Comment by chrisjsewell** [comment]
> Note that eqnarray is the default way to write systems of equations in latex

Added in 9d22992, for users who want to be stupid 😉

**Comment by chrisjsewell** [comment]
> I see flalign on your list, but it doesn't look like it's an environment either mathjax or katex support.

Yeh strange that they don't. I think it should be kept here, since the goal is to capture all top-level amsmath environments, then handle it in some way as you mention upstream in myst-parser.
There are open issue for it on both repos:
- https://github.com/KaTeX/KaTeX/issues/1990
- https://github.com/mathjax/MathJax/issues/1075

Note a another thing to consider, which again mathjax and katex don't currently support but are good for latex PDF, would be the environments in http://mirror.ox.ac.uk/sites/ctan.org/macros/latex/contrib/breqn/breqn.pdf. Probably add this as an enhancement issue somewhere in executablebooks

**Comment by phaustin** [comment]
Makes sense. How much effort will it be to handle something like https://mhchem.github.io/MathJax-mhchem ?

**Comment by choldgraf** [comment]
way to go!

**Pull request merged**